src/BlockTypes/ProductGallery.php

```diff
@@ -121,12 +121,13 @@ protected function render( $attributes, $content, $block ) {
 			$classname_single_image = 'is-single-product-gallery-image';
 		}
 
-		$number_of_thumbnails = $block->attributes['thumbnailsNumberOfThumbnails'] ?? 0;
-		$classname            = $attributes['className'] ?? '';
-		$dialog               = isset( $attributes['mode'] ) && 'full' !== $attributes['mode'] ? $this->render_dialog() : '';
-		$post_id              = $block->context['postId'] ?? '';
-		$product              = wc_get_product( $post_id );
-		$product_id           = strval( $product->get_id() );
+		$number_of_thumbnails           = $block->attributes['thumbnailsNumberOfThumbnails'] ?? 0;
+		$classname                      = $attributes['className'] ?? '';
+		$dialog                         = isset( $attributes['mode'] ) && 'full' !== $attributes['mode'] ? $this->render_dialog() : '';
+		$post_id                        = $block->context['postId'] ?? '';
+		$product                        = wc_get_product( $post_id );
+		$product_gallery_first_image_id = reset( ProductGalleryUtils::get_product_gallery_image_ids( $product, 1 ) );
```

Comment:
`ProductGalleryUtils::get_product_gallery_image_ids( $product, 1 )` has to be assigned to a variable first before using `reset()` on it otherwise you will get "pass by reference warning".